sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -104,6 +105,9 @@ trait CheckAnalysis {
               s"filter expression '${f.condition.prettyString}' " +
                 s"of type ${f.condition.dataType.simpleString} is not a boolean.")
 
+          case j @ Join(_, _, NaturalJoin(_), _) =>
+            failAnalysis(s"natural join not resolved.")
```

Comment:
This will never happen but when there were remaining natural joins after 100 iterations